### PR TITLE
fix: keep classes for string and base spacings

### PR DESCRIPTION
**Pull request by sashuk**
[FX-4262]

### Description

This pull request
- restores original implementation of strings constants (`xsmall`, `small`, etc.) as Container spacing properties values. This is needed to ensure the same representation of components after consumers update to the picasso version with BASE spacings.
- implements Container spacing properties in a way, that when BASE spacing values are provided, CSS classes are also used to ease the migration to TailwindCSS

**What is the problem?**

Let's take`top` and `bottom` spacing properties as an example

- `master` branch implementation creates dedicated CSS class if spacing is provided as string constant. If it is provided as number, then the `style` attribute is added to the container

Screenshot 2023-09-22 at 18 44 32

- https://github.com/toptal/picasso/tree/feature/base-spacing branch implementation treats any spacing (string constant, number, or BASE spacing) the same way and adds the `style` attribute to the container

Screenshot 2023-09-22 at 18 49 12

❗ The problem is that in rare cases consumers rely on the fact that **string constants spacing is set via CSS classes**, so they implement custom styling via CSS classes that is supposed to overwrite the CSS classes coming from Picasso. The example above has the custom CSS styling as below, so it stops working if the implementation via only `style` attribute is kept.

Screenshot 2023-09-22 at 18 52 08

### How to test

- [Temploy](https://picasso.toptal.net/FX-4262-restore-classes)
- Compare same story in https://picasso.toptal.net/fx-4262-merge-base-spacing-into-master-and-announce-the-change/?path=/story/components-timeline--timeline and https://picasso.toptal.net/FX-4262-restore-classes/?path=/story/components-timeline--timeline, it should have different bottom padding (as highlighted in example) and should be the same as `master` https://picasso.toptal.net/?path=/story/components-timeline--timeline

### Development checks

- [N/A] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [N/A] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [N/A] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4262]: https://toptal-core.atlassian.net/browse/FX-4262?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

**Review by dmaklygin — APPROVED** [review]

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

File: jest.spec.mjs [hunk 1]
```diff
@@ -11,6 +11,10 @@ const config = {
     '^@toptal/picasso$': '<rootDir>/packages/picasso/src/index.ts',
     '^@toptal/picasso-provider$':
       '<rootDir>/packages/picasso-provider/src/index.ts',
+    '^@toptal/picasso-provider/utils$':
```
Comment (sashuk) [inline jest.spec.mjs:14]: Needed for unit tests configuration

File: packages/picasso-provider/src/Picasso/config/spacings.ts [hunk 2]
```diff
@@ -36,13 +36,19 @@ export enum SpacingEnum {
 
 class PicassoSpacing {
   #value: PicassoSpacingValues
+  #index: number
```
Comment (sashuk) [inline packages/picasso-provider/src/Picasso/config/spacings.ts:39]: Needed for indexing of spacings, i.e. `PicassoSpacing` should be aware of what is the index of corresponding `SPACING_INDEX` constant

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Review by TomasSlama — APPROVED** [review]
good job!

**Pull request merged**